### DISCUSSION TITLE
file_dialog_open returns empty string

### DISCUSSION THREAD

**Discussion by jcxz100h** [libigl/libigl#2272] [Q&A] [answered]
Dear all,

I am trying to use `igl::file_dialog_open` to get a file path from the user, but **sometimes**, it returns an empty string:

```
    if (ImGui::Button("Load mesh", ImVec2(-1, 0)))
    {
      // load a mesh
      std::string fn = igl::file_dialog_open();
      fmt::print("dialog returned {}\n", fn);
      
      if (!fn.empty()) // <------ NEVER TRUE
      {
        fmt::print("selected file {}\n", fn);
        meshes[show_mesh] = reader(fn);

        if (meshes[show_mesh].isloaded())
        {
          viewer.data().set_mesh(meshes[show_mesh].verts, meshes[show_mesh].faces);
          fmt::print("mesh loaded correctly for {}\n", (int)show_mesh);
        }
      }
      else
        fmt::print("load canceled\n");
    }
```

Am I missing something? I am running this on macOS, probably this matters.

Thanks!

**Comment by alecjacobson** [comment, marked as answer]
Oof. It's been a long time since I used this. I think there are other options for universal file dialogs out there. The mac implementation is running a system command to an applescript attached to some bash command line junk. It worked at the time and not sure why it'd return empty string, but it sure is a hacky way to get a file name.

**Reply by jcxz100h** [reply]
Thanks for your answer! It will be sufficient for now, don't worry!
